### PR TITLE
Provided CMS previews for all collections and pages

### PR DESCRIPTION
**Pull request by rb1193**
@williamparry suggested this would be valuable in helping content managers do their work. It will require some upkeep when adding new collections and changing component names but hopefully that won't happen too often.

Sorry it's a bit of a messy diff, my linter picked a lot of stuff up on some of the older pages.
___
NB - by making a pull request you confirm agreement to the [Contributors Agreement](https://github.com/openlawnz/openlawnz-web/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: src/cms/previews.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import Wizard from "../components/Wizard/Wizard"
+import { AboutPageContent } from "../pages/about"
+import { DevelopersPageContent } from "../pages/developers"
+import { GetInvolvedPageContent } from "../templates/get-involved-page"
+import { MicrositeContent } from "../templates/microsite"
+import { SingleNewsContent } from "../templates/single-news"
+import { OurMissionPageContent } from "../templates/our-mission-page"
+
+export function aboutPreview({ entry }) {
+  const data = entry.get("data")
```
Comment (danieljimeneznz) [inline src/cms/previews.js:13]: What type is entry?

Comment (rb1193) [reply src/cms/previews.js:13]: An immutable map representation of a netlify cms entry (the about page, in this case). I should comment stuff more

Comment (danieljimeneznz) [reply src/cms/previews.js:13]: Thanks for the clarification, really should've read the docs on `registerPreviewTemplate` beforehand 😄

Comment (danieljimeneznz) [reply src/cms/previews.js:13]: Could actually do `const data = toObject(entry.get("data"))`
then your return would look like
```
  const { content, directors, contributors } = data;
  return (
    <AboutPageContent
      content={content}
      directors={directors}
      contributors={contributors}
    />
  )

File: src/cms/previews.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+export function ourMissionPreview({ entry }) {
+  return <OurMissionPageContent pageContext={toObject(entry.get("data"))} />
+}
+
+export function pluginPreview({ entry }) {
+  return <PluginShowcase data={[toObject(entry.get("data"))]} />
+}
+
+export function micrositesPreview({ entry }) {
+  const immutableData = entry.get("data")
```
Comment (danieljimeneznz) [inline src/cms/previews.js:45]:
```suggestion
  const data = toObject(entry.get("data"))
```

File: src/cms/previews.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      content={toArray(data.get("content"))}
+      directors={toObject(data.get("directors"))}
+      contributors={toArray(data.get("contributors"))}
+    />
+  )
+}
+
+export function developersPreview({ entry }) {
+  const data = toObject(entry.get("data"))
+  return <DevelopersPageContent content={data} />
```
Comment (danieljimeneznz) [inline src/cms/previews.js:25]:
```suggestion
  return <DevelopersPageContent content={toObject(entry.get("data")} />
```

File: src/cms/previews.js (hunk 1 above)
Comment (danieljimeneznz) [reply src/cms/previews.js:13]: `const { data: { content, directors, contributors } } = entry.toJS();`

**Comment by rb1193** [comment]
@danieljimeneznz I think this should be a bit more readable now, if you wouldn't mind taking a look?

**Pull request merged**